hibernate-core/src/main/java/org/hibernate/type/ComponentType.java

```diff
@@ -176,9 +176,7 @@ public boolean isEqual(final Object x, final Object y) throws HibernateException
 		if ( x == y ) {
 			return true;
 		}
-		if ( x == null || y == null ) {
```

Comment:
Ok that makes sense, to make the handling reflexive.  I'm ok with keeping this.

Could you please look into the conflicts though.  Thanks